tests/EventListener/PayPalV2ExpressEventListenerTest.php

```diff
@@ -0,0 +1,224 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PayonePayment\EventListener;
+
+use PayonePayment\Components\ConfigReader\ConfigReaderInterface;
+use PayonePayment\Components\Helper\ActivePaymentMethodsLoaderInterface;
+use PayonePayment\PaymentMethod\PayonePaypalV2Express;
+use PayonePayment\Storefront\Struct\PayPalV2ExpressButtonData;
+use PayonePayment\Struct\Configuration;
+use PayonePayment\TestCaseBase\PayoneTestBehavior;
+use PHPUnit\Framework\TestCase;
+use Psr\Log\LoggerInterface;
+use Shopware\Core\Framework\Uuid\Uuid;
+use Shopware\Storefront\Page\Checkout\Cart\CheckoutCartPage;
+use Shopware\Storefront\Page\Checkout\Cart\CheckoutCartPageLoadedEvent;
+use Shopware\Storefront\Page\Checkout\Offcanvas\OffcanvasCartPage;
+use Shopware\Storefront\Page\Checkout\Offcanvas\OffcanvasCartPageLoadedEvent;
+use Shopware\Storefront\Page\Checkout\Register\CheckoutRegisterPage;
+use Shopware\Storefront\Page\Checkout\Register\CheckoutRegisterPageLoadedEvent;
+use Shopware\Storefront\Page\Page;
+use Shopware\Storefront\Page\PageLoadedEvent;
+use Symfony\Component\Routing\RouterInterface;
+
+/**
+ * @covers \PayonePayment\EventListener\PayPalV2ExpressEventListener
+ */
+class PayPalV2ExpressEventListenerTest extends TestCase
+{
+    use PayoneTestBehavior;
+
+    /**
+     * @dataProvider subscribedEventClasses
+     * @testdox It adds PayPalV2ExpressButtonData extension on $pageClass
+     *
+     * @param class-string<PageLoadedEvent> $eventClass
+     * @param class-string<Page> $pageClass
+     */
+    public function testItAddsExpressButtonDataExtension(string $eventClass, string $pageClass): void
+    {
+        $activePaymentMethodIds = [
+            Uuid::randomHex(),
+            PayonePaypalV2Express::UUID,
+        ];
+
+        $salesChannelContext = $this->createSalesChannelContextWithLoggedInCustomerAndWithNavigation();
+        $salesChannelContext->getCurrency()->setIsoCode('EUR');
+
+        $request = $this->getRequestWithSession([]);
+        $request->setLocale('de-DE');
+
+        $configuration = new Configuration([
+            'transactionMode' => 'test',
+            'paypalV2ExpressPayPalMerchantId' => 'the-merchant-id',
+            'paypalV2ExpressShowPayLaterButton' => false,
+        ]);
+
+        $activePaymentMethodsLoader = $this->createMock(ActivePaymentMethodsLoaderInterface::class);
+        $activePaymentMethodsLoader
```

Comment:
please keep the testcase simple. 
remove the `expect` and the `with`

we wont expect any method calls. it is only required to make sure, the method has a return-value. And if the return value does not change between different runs, we only need to have
```php
$activePaymentMethodsLoader->method('getActivePaymentMethodIds')->willReturn($activePaymentMethodIds);
```

Reply:
please also apply this to the other statements/testcases

Reply:
Done